src/query/print.go

```diff
@@ -10,12 +10,15 @@ import (
 	"time"
 
 	"github.com/thought-machine/please/src/core"
+	"github.com/thought-machine/please/src/parse"
 )
 
 // Print produces a Python call which would (hopefully) regenerate the same build rule if run.
 // This is of course not ideal since they were almost certainly created as a java_library
 // or some similar wrapper rule, but we've lost that information by now.
-func Print(graph *core.BuildGraph, targets []core.BuildLabel, fields, labels []string) {
+func Print(state *core.BuildState, targets []core.BuildLabel, fields, labels []string) {
+	graph := state.Graph
+	order := parse.NewAspParser(state).BuildRuleArgOrder()
```

Comment:
Why do you need to create a new parser? Can't you just use `state.Parser` to answer this question?

Reply:
I'd have to cast it to `parser.aspParser` and then get at the internal `asp.Parser` field on that. We don't expose any of that so I'd have to expose more things. I think you're right, that's probably the better option.